tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedAssertSessionPinnedOperation.cs

```diff
@@ -0,0 +1,66 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Tests.Specifications.unified_test_format;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public class UnifiedAssertSessionPinnedOperation : IUnifiedSpecialTestOperation
+    {
+        private readonly IClientSessionHandle _session;
+
+        public UnifiedAssertSessionPinnedOperation(IClientSessionHandle session)
+        {
+            _session = session;
+        }
+
+        public void Execute()
+        {
+            _session.WrappedCoreSession?.CurrentTransaction?.PinnedServer.Should().NotBeNull();
```

Comment:
I think this should be:

            _session.WrappedCoreSession.CurrentTransaction.PinnedServer.Should().NotBeNull();

otherwise a null CurrentTransaction will result in the test passing when it should not.

Note: WrappedCoreSession will never be null.